tools/misc/update_model_index.py

```diff
@@ -1,10 +1,47 @@
-import argparse
+#!/usr/bin/env python
+
+# This tool is used to update model-index.yml which is required by MIM, and
+# will be automatically called as a pre-commit hook. The updating will be
+# triggered if any change of model information (.md files in configs/) has been
+# detected before a commit.
+
 import os
 import os.path as osp
 import re
+import sys
 
 import mmcv
 
+MMPOSE_ROOT = osp.dirname(osp.dirname(osp.dirname(__file__)))
+
+
+def dump_yaml_and_check_difference(obj, file):
+    """Dump object to a yaml file, and check if the file content is different
+    from the original.
+
+    Args:
+        obj (any): The python object to be dumped.
+        file (str): YAML filename to dump the object to.
+    Returns:
+        Bool: If the target YAML file is different from the original.
+    """
+
+    original = None
+    if osp.isfile(file):
+        with open(file, 'r') as f:
```

Comment:
Better to explicitly open with utf8 encoding

Reply:
fixed.